src/utils.c

```diff
@@ -1865,9 +1865,12 @@ relation_set_reloption_impl(Relation rel, List *options, LOCKMODE lockmode)
 
 	Relation pgclass = table_open(RelationRelationId, RowExclusiveLock);
 	Oid relid = RelationGetRelid(rel);
-	HeapTuple tuple = SearchSysCache1(RELOID, ObjectIdGetDatum(relid));
+	HeapTuple tuple = SearchSysCacheLockedCopy1(RELOID, ObjectIdGetDatum(relid));
```

Comment:
Since you're creating a copy here, shouldn't you have a matching call for `heap_freetuple`?

Reply:
Yes.